### DISCUSSION TITLE
让简悦完美支持本地文件并可加入稍后读

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#4519] [使用技巧]
# 重要提示

> 嗨，[同步助手 1.1.3 版](https://github.com/Kenshin/simpread/discussions/4049#discussioncomment-4192563) 内置了此功能，详细说明 [稍后读 · 读取外部文件](https://github.com/Kenshin/simpread/discussions/4625)。

***

# 简悦支持打开本地文件

简悦支持打开本地文件并进入阅读模式，但因 Chrome 的限制，所以需要实现开启相关选项，操作细节 [请看这里](https://github.com/Kenshin/simpread/discussions/1689)。

# 不支持本地文件系统加入稍后读

如果你使用了 **同步助手的自动同步功能，在本地文件中即便进入了阅读模式，也没法加入稍后读**，加入后会有下图所示的提示：

![image](https://user-images.githubusercontent.com/81074/191930085-1661b1a3-78ff-42ae-b9b5-4bde5c0761e8.png)

如果你没有使用同步助手的自动同步功能，虽然**可以正常的加入稍后读，但却没有本地快照**。

> 接下来介绍一个非常简单的方式即可让简悦支持本地文件且可加入稍后读，自然也包含 `output` 本地快照。

# 简悦读取本地文件并加入稍后读的完美方案

利用一个 Chrome 应用：**Web Server for Chrome** 就能实现此需求。

## 下载地址

https://chrome.google.com/webstore/detail/web-server-for-chrome/ofhbbkphhbklhfoeikjpcbhemlocgigb

## 安装

下载并安装后，会出现在 `chrome://extensions/`

![image](https://user-images.githubusercontent.com/81074/191930536-66e7f6b1-8647-44e1-83c7-a28358168413.png)

## 使用

打开 `chrome://apps/` 就能看到这个 Chrome 应用

![image](https://user-images.githubusercontent.com/81074/191930702-fab27828-8054-4a86-8d69-97e0ca037a7d.png)

点击打开它

<img width="370" alt="image" src="https://user-images.githubusercontent.com/81074/191930783-f894559f-bfe3-4749-a4e1-9805f2e87345.png">

## 设置

基本上不需要做任何设置，需要指定一个本地文件夹即可。

<img width="370" alt="image" src="https://user-images.githubusercontent.com/81074/191931851-a262b48c-a202-48f2-b355-ee9c016882bb.png">

注意：这里指定的文件夹最好是存放本地且需要加入到简悦中的文件。

## 使用

打开 http://127.0.0.1:8887 后就能看到当前文件夹对应的本地文件（截图中的文件只是参考，与你选择后的文件夹里面的内容为主）

<img width="900" alt="image" src="https://user-images.githubusercontent.com/81074/191946401-33bd1446-bc7b-4ffb-8dcb-e78483b0904d.png">

## 视频

https://user-images.githubusercontent.com/81074/191948063-f5278551-c0ba-47a2-b844-8cd2e204ed36.mp4

**Comment by TYZRPVX** [comment]
这个功能有点厉害。

对于一些可能 404 的文章，保存 html 相比保存 pdf 通常更加能还原格式，pdf 类文档批注软件相对来说更丰富，但 html 批注就很少了，这个方法完美解决这个问题

**Reply by Kenshin** [reply]
哈，是的。

只不过目前使用了第三方工具，接下来会考虑简悦直接内置这个方案。

**Reply by Kenshin** [reply]
@TYZRPVX 

[同步助手 1.1.3 版](https://github.com/Kenshin/simpread/discussions/4049#discussioncomment-4192563) 内置了此功能，详细说明 [稍后读 · 读取外部文件](https://github.com/Kenshin/simpread/discussions/4625)。

**Comment by Chengjiuyou** [comment]
安装扩展后提示"Web Server for Chrome"不再受支持，该怎么办呢？

**Reply by Kenshin** [reply]
仔细看本页最开头... 😂

**Comment by takeiteasyman** [comment]
作者您好！请问删除一条带本地缓存的稍后读，本地缓存的html文件只能手动删除么？

**Reply by Kenshin** [reply]
目前在稍后读中删除是这样的，但你可以使用 [稍后读 · 极速版](https://github.com/Kenshin/simpread/discussions/3864) 来删除。

**Reply by takeiteasyman** [reply]
感谢您的回复，但我没有在稍后读-极速版 中找到删除功能

**Reply by Kenshin** [reply]
直接删除这里快照会自动同步删除。
 
<img width="333" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/b22288db-0f79-4cca-ba94-c291cf03a209">

**Reply by Kenshin** [reply]
> 作者您好！感谢你的截图，让问题清晰了。这是我这边的截图，确实缺少了删除按钮，这算是发现了bug么？这种情况应该怎么处理呢
![image](https://github.com/Kenshin/simpread/assets/45447994/61da6277-3c62-4b80-8a7d-34376605817c)

鼠标移上的话，**如果你那边确认没有的话，可能是我这边是新版本**（还没有发布呢） 😂

**Reply by takeiteasyman** [reply]
好的，谢谢你的回复，期待新版本的到来。其次，手动删掉某个缓存后，output文件序号会出现断层，这个是正常的么？我个人对序号不连续是可以接受的

**Reply by Kenshin** [reply]
> 手动删掉某个缓存后，output文件序号会出现断层，这个是正常的么

是的。从某种角度上说，删除的 `ID` 也是它存在过的「证据」。😄
